### PR TITLE
Add v3 Transaction Types

### PR DESCRIPTION
**Pull request by cdt-eth**

### CODE REVIEW

File: src/types/transaction.ts [hunk 1]
```diff
@@ -172,3 +172,57 @@ export type RawNeighbor = {
...
+  assetsEnriched?: { [key: string]: AssetEnriched };
+};
+ 
+export type TransactionEnriched = {
+  parties: string[];
+  context: TransactionContextType;
+  netAssetTransfers: {
+    [address: string]: {
+      received: [{ type; contract; value; tokenId? }];
+      sent: [{ type; contract; value; tokenId? }];
```
Comment (guscost) [inline src/types/transaction.ts:190]: Could we add the type of each of these sub-object properties as well as the property names?

File: src/types/transaction.ts [hunk 2]
```diff
@@ -172,3 +172,57 @@ export type RawNeighbor = {
...
+} & Transaction;
+
+export type PartyEnriched = {
+  chainId: number;
+  label: {
+    public: string;
+  };
+  isContract: boolean;
+  tokenStandard: string;
+  imgUrl: string;
```
Comment (guscost) [inline src/types/transaction.ts:204]: We use `imageUrl` below, but I suppose this has to be accurate, maybe call it out with a comment until we can update.

Also I would add comments stating explicitly that all these structures are in progress, like https://github.com/Once-Upon/docs/pull/22/files

**Review by guscost — APPROVED** [review]

File: src/types/transaction.ts (hunk 2 above)
Comment (cdt-eth) [reply src/types/transaction.ts:231]: https://github.com/Once-Upon/context/pull/382/commits/a9e66a8b1a38b3c7ce3642304c1f131ed6d9f324#diff-dc1174f013a1e9b407e473204d38c757aa935a46734f178214e63e2697613e76R176-R178

File: src/types/transaction.ts (hunk 1 above)
Comment (cdt-eth) [reply src/types/transaction.ts]: [added sub-types](https://github.com/Once-Upon/context/pull/382/commits/a9e66a8b1a38b3c7ce3642304c1f131ed6d9f324#diff-dc1174f013a1e9b407e473204d38c757aa935a46734f178214e63e2697613e76R199-R223)

File: src/types/transaction.ts [hunk 3]
```diff
@@ -203,15 +203,6 @@ type TransferDetails = {
...
 
-type AssetTransfer = {
-  from: string;
-  to: string;
-  value: string;
-  type: string;
-  contract: string;
-  tokenId?: string;
-};
-
```
Comment (cdt-eth) [inline src/types/transaction.ts]: didn't realize we already had a type for this

**Review by guscost — APPROVED** [review]

**Pull request merged**